### PR TITLE
New package: Fossil.Fossil version 2.21

### LINKED ISSUE #96961: [Package Request]: Fossil
### Package Requested

- [X] I would like someone else to build the manifest.
- [ ] I would like help so I can submit the manifest.
- [X] I have performed a search and couldn't find this package.
- [X] I have checked that the installer for this package is not a script file
- [ ] I think there is a new version available and I have provided the URL.

### Please provide the following information

Publisher: Fossil
Publisher Url: https://fossil-scm.org
Package Name: Fossil
Description: A distributed version control system with project management capabilities and a web interface.
Package Version: 2.20
Installer URL: https://fossil-scm.org/home/uv/download.html

Current Package Identifier (for new versions): N/A

### PR DESCRIPTION
**Pull request by mdanish-kh**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----
- Resolves #96961
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/96974)

### CODE REVIEW

**Review by ItzLevvie — DISMISSED** [review]

File: manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+PackageVersion: "2.20"
+InstallerType: zip
+Installers:
+- Architecture: x64
+  NestedInstallerType: portable
+  NestedInstallerFiles:
+  - RelativeFilePath: .\fossil.exe
+    PortableCommandAlias: fossil
+  InstallerUrl: https://fossil-scm.org/home/uv/fossil-w64-202302092100.zip
+  InstallerSha256: 4C9B56CD1AA3E3AC13478325B15E2EB36DBCEB3E13BB4F8D94BA055D73139F27
```
Comment (rodrigoborgesdeoliveira) [inline manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml:14]: The x64 file has been updated.
```suggestion
  InstallerSha256: E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
```

File: manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- Architecture: x86
+  NestedInstallerType: portable
+  NestedInstallerFiles:
+  - RelativeFilePath: .\fossil.exe
+    PortableCommandAlias: fossil
+  InstallerUrl: https://fossil-scm.org/home/uv/fossil-w32-202302092100.zip
+  InstallerSha256: C96195CA402F51FA6650C3241E952699A3AAEB26491114D21DDFC883299DDEF4
```
Comment (rodrigoborgesdeoliveira) [inline manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml:21]: This is somehow matching the x64's hash.
```suggestion
  InstallerSha256: E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
```

**Review by rodrigoborgesdeoliveira — CHANGES_REQUESTED** [review]

File: manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml (hunk 1 above)
Comment (mdanish-kh) [reply manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml:14]: Not sure about you, but I'm getting a 0 bytes downloaded .zip file with the hash you suggested 👀 Seems like an issue with the Url

Comment (mdanish-kh) [reply manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml:14]: Looks like there's a new version available `2.21` and prev Url is not working

**Review by stephengillie — APPROVED** [review]

File: manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml (hunk 1 above)
Comment (stephengillie) [reply manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml]: The hash 4C9...F27 works and has been manually validated. Where is the hash E3B...855 seen?

Comment (mdanish-kh) [reply manifests/f/Fossil/Fossil/2.20/Fossil.Fossil.installer.yaml]: It was in the previous commit https://github.com/microsoft/winget-pkgs/pull/96974/commits/6d05ecafe89b01216c8d2165bf9afcfd5301c366 when version `2.20` was still around and was the latest one.

**Pull request merged**